src/libpmemobj/cuckoo.c

```diff
@@ -55,7 +55,7 @@ struct cuckoo_slot {
 
 struct cuckoo {
 	size_t size; /* number of hash table slots */
-	struct cuckoo_slot *tab;
+	struct cuckoo_slot *restrict tab;
```

Comment:
Hmm, I briefly checked the code, and immediately I found once place in which this breaks down `struct cuckoo_slot *oldtab = c->tab;` in `cuckoo_grow`.

I general, I don't like using `restrict` - the benefit is questionable and we are introducing some *severe* limitations that the compiler does not warn us about breaking.